common/src/main/java/com/microsoft/hydralab/performance/PerformanceTestManagementService.java

```diff
@@ -95,19 +109,140 @@ private ITestRun getTestRun() {
+        testRunPerfResultMap.remove(testRun.getId());
+
+        //TODO Android battery: adb shell dumpsys battery reset using Device info
+    }
+
+    private void inspectWithLifeCycle(InspectionStrategy.WhenType whenType, String description) {
+        List<InspectionStrategy> strategyList = testLifeCycleStrategyMap.get(getTestRun().getId());
+        if (strategyList == null) return;
+
+        for (InspectionStrategy inspectionStrategy : strategyList) {
+            if (inspectionStrategy == null) continue;
+            PerformanceInspection inspection = inspectionStrategy.inspection;
+            if (inspectionStrategy.when == null) continue;
+
+            if (inspectionStrategy.when.contains(whenType) || whenType == InspectionStrategy.WhenType.TEST_RUN_STARTED) {
+                PerformanceInspection lifeCycleInspection = new PerformanceInspection(
+                        whenType.name() + "-" + description, inspection.inspectorType, inspection.appId,
+                        inspection.deviceIdentifier, whenType == InspectionStrategy.WhenType.TEST_RUN_STARTED);
+                inspect(lifeCycleInspection);
+            }
+        }
+    }
+
+    /**
+     * For giving inspection return the inspection with device id that related to test run
+     */
+    private PerformanceInspection getDevicePerformanceInspection(PerformanceInspection inspection) {
+        return new PerformanceInspection(inspection.description, inspection.inspectorType, inspection.appId,
+                // For windows inspector, the deviceIdentifier is useless
+                getTestRun().getDeviceSerialNumber(), inspection.isReset);
+    }
+
+    private List<PerformanceTestResult> parseForTestRun(ITestRun testRun) {
+        Map<String, PerformanceTestResult> testResultMap = testRunPerfResultMap.get(testRun.getId());
+        if (testResultMap == null) return null;
+
+        List<PerformanceTestResult> resultList = new ArrayList<>();
+        for (PerformanceTestResult performanceTestResult : testResultMap.values()) {
+            PerformanceResultParser parser = getParserByType(performanceTestResult.parserType);
+            Assert.notNull(parser, "Found no matched result parser: " + performanceTestResult.parserType);
+            resultList.add(parser.parse(performanceTestResult));
+        }
+        return resultList;
+    }
+
+    private void savePerformanceTestResults(List<PerformanceTestResult> resultList, Logger log) {
+        //TODO save results to DB
+        if (resultList != null && !resultList.isEmpty()) {
+            try {
+                FileUtil.writeToFile(new ObjectMapper().writeValueAsString(resultList),
+                        new File(getTestRun().getResultFolder(), PerformanceInspection.class.getSimpleName())
+                                + File.separator + "PerformanceReport.json");
+            } catch (JsonProcessingException e) {
+                log.error("Failed to save performance test results to file");
```

Comment:
The exception is ignored,  at least we can log it to provide clarify when we have to resort to the log.
```suggestion
                log.error("Failed to save performance test results to file", e);
```

Reply:
More about handling exception: https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#handling-exceptions-in-code

Reply:
Updated